formidling/src/main/java/no/nav/ung/sak/formidling/bestilling/VurderVedtaksbrevTask.java

```diff
@@ -0,0 +1,170 @@
+@FagsakProsesstaskRekkefølge(gruppeSekvens = false)
+public class VurderVedtaksbrevTask extends BehandlingProsessTask {
+
+    private static final Logger LOG = LoggerFactory.getLogger(VurderVedtaksbrevTask.class);
+
+    public static final String TASKTYPE = "formidling.vedtak.brevvurdering";
+
+    private boolean enableIgnoreManglendeBrev;
+    private VedtaksbrevRegler vedtaksbrevRegler;
+    private ProsessTaskTjeneste prosessTaskTjeneste;
+    private VedtaksbrevValgRepository vedtaksbrevValgRepository;
+    private BrevbestillingTjeneste brevbestillingTjeneste;
+    private VedtaksbrevResultatRepository vedtaksbrevResultatRepository;
+    private BehandlingRepository behandlingRepository;
+
+    VurderVedtaksbrevTask() {
+        // for proxy
+    }
+
+    @Inject
+    public VurderVedtaksbrevTask(
+        VedtaksbrevRegler vedtaksbrevRegler,
+        @KonfigVerdi(value = "IGNORE_MANGLENDE_BREV", defaultVerdi = "false") boolean ignoreManglendeBrev,
+        ProsessTaskTjeneste prosessTaskTjeneste,
+        VedtaksbrevValgRepository vedtaksbrevValgRepository,
+        BrevbestillingTjeneste brevbestillingTjeneste,
+        VedtaksbrevResultatRepository vedtaksbrevResultatRepository, BehandlingRepository behandlingRepository) {
+
+        this.enableIgnoreManglendeBrev = ignoreManglendeBrev;
+        this.vedtaksbrevRegler = vedtaksbrevRegler;
+        this.prosessTaskTjeneste = prosessTaskTjeneste;
+        this.vedtaksbrevValgRepository = vedtaksbrevValgRepository;
+        this.brevbestillingTjeneste = brevbestillingTjeneste;
+        this.vedtaksbrevResultatRepository = vedtaksbrevResultatRepository;
+        this.behandlingRepository = behandlingRepository;
+    }
+
+    @Override
+    protected void prosesser(ProsessTaskData prosessTaskData) {
+        Long behandlingId = Long.valueOf(prosessTaskData.getBehandlingId());
+        var behandling = behandlingRepository.hentBehandling(behandlingId);
+
+        var resultat = vedtaksbrevRegler.kjør(behandlingId);
+        LOG.info("Resultat fra vedtaksbrev regler: {}", resultat.safePrint());
+
+        if (!resultat.harBrev()) {
+            håndterIngenBrevResultat(resultat, behandling);
+            return;
+        }
+
+        var vedtaksbrevValgEntitet = vedtaksbrevValgRepository.finnVedtakbrevValg(behandlingId).orElse(null);
+        boolean harSaksbehandlerHindretEllerRedigertBrev = vedtaksbrevValgEntitet != null && (vedtaksbrevValgEntitet.isHindret() || vedtaksbrevValgEntitet.isRedigert());
+        if (harSaksbehandlerHindretEllerRedigertBrev) {
+            håndterSaksbehandlerValg(behandling, vedtaksbrevValgEntitet, resultat);
+            return;
+        }
+
+        if (vedtaksbrevValgEntitet != null) {
+            LOG.warn("Vedtaksbrevvalg lagret, men verken hindret eller redigert");
+        }
+
+        var bestillinger = resultat.vedtaksbrevResultater()
+            .stream()
+            .map(it -> lagreVedtaksbrevResultatOgOpprettBestilling(behandling, it))
```

Comment:
Eg er ikkje så fan av map med side effects. Då synes eg det er bedre å mappe for så å kjøre foreach